From 9f62fc907385b6ddf546dae933519bd4da81e984 Mon Sep 17 00:00:00 2001
From: Alexandra Zota <zota@adobe.com>
Date: Tue, 21 Feb 2023 11:19:37 +0200
Subject: [PATCH] ACP2E-1648: fix magnifier loading

---
 lib/web/magnifier/magnifier.js | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/lib/web/magnifier/magnifier.js b/lib/web/magnifier/magnifier.js
index 29726688f4d5f..8b525b2e047b8 100644
--- a/lib/web/magnifier/magnifier.js
+++ b/lib/web/magnifier/magnifier.js
@@ -541,11 +541,9 @@
 
             showWrapper = false;
             $(thumbObj).on('load', function () {
-                if (data.length > 0) {
+                if (data.hasOwnProperty(idx)) {
                     data[idx].status = 1;
-
                     $(largeObj).on('load', function () {
-
                         if (largeObj.width > largeWrapper.width() || largeObj.height > largeWrapper.height()) {
                             showWrapper = true;
                             bindEvents(eventType, thumb);
@@ -559,7 +557,6 @@
                             updateLensOnLoad(idx, thumb, largeObj, largeWrapper);
                         }
                     });
-
                     largeObj.src = data[idx].largeUrl;
                 }
             });
